### ISSUE TITLE
Stringtie2 installation: undefined reference to _intel_fast_memset'

### ISSUE THREAD

**Issue by NatPRoach** [gpertea/stringtie#228]
Hello, sorry to open another issue about installation so quickly, but I'm attempting to install Stringtie2 on a different machine (a Linux machine running CentOS 7) than the previous issue, and I've run in to an error I was hoping you may be able to help with. Running `make release` on this machine results in a bunch of error messages of the form `undefined reference to _intel_fast_memset`. They don't all say `_intel_fast_memset`, but they all take the form `undefined reference to _intel_*`.
I tried looking up the error message, but didn't find any recommended fixes at the step of running `make release`. I'd appreciate any help you can provide with regards to this issue. Thanks in advance.

**Comment by gpertea** [comment]
Is it possible that the current/default C++ compiler is not GCC's g++ on that machine (and it could be the Intel's compiler instead?). What does the ` g++ --version` command output looks like ? Or perhaps CXX is set to a different compiler (because StringTie's Makefile is going to use CXX if set). Does `echo $CXX` return something else than g++ (or it is empty?) ?

**Comment by NatPRoach** [comment]
This is a computational cluster with the ability to load software as modules, so by default the result of `echo $CXX` is `icpc` and the result of `g++ --version` is `g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)`, but if I load gcc as a module using `module load gcc/6.4.0` the result of `echo $CXX` is `g++` and the result of `g++ --version` is `g++ (GCC) 6.4.0`. Both configurations result in the error messages described above upon running `make release`.

**Comment by gpertea** [comment]
Sounds similar to a problem I've encountered on MARCC a while ago (I guess that's where you're trying this too?).  Can you run `module unload intel` first (in a new session if possible, or unload any other modules like gcc/6.4.0 if you loaded them).  $CXX should then be unset and the system gcc/g++ should kick in. The default gcc 4.8.5 of CentOS 7 should work just fine, I don't know why the MARCC admins decided that the default should be the Intel compiler instead, for everyone..

**Comment by gpertea** [comment]
Ah I forgot - please run `make clean` before `make release` (or use `make clean release`) after changing any compiler settings. Loading the module gcc/6.4.0 as you tried earlier should've worked, but the object files from the previous build attempt were in the way, `make clean` would've allowed you to start fresh there.

**Comment by NatPRoach** [comment]
Yes I'm working on MARCC, that seems to have done the trick. Thanks for the quick response!
